apps/taxes/src/modules/taxes/active-tax-provider.ts

```diff
@@ -1,45 +1,113 @@
-import { TaxBaseFragment } from "../../../generated/graphql";
+import {
+  MetadataItem,
+  OrderCreatedSubscriptionFragment,
+  OrderFulfilledSubscriptionFragment,
+  TaxBaseFragment,
+} from "../../../generated/graphql";
 import { createLogger } from "../../lib/logger";
 import { ChannelConfig } from "../channels-configuration/channels-config";
 import { ProviderConfig } from "../providers-configuration/providers-config";
-import { AvataxProvider } from "../avatax/avatax-provider";
-import { TaxJarProvider } from "../taxjar/taxjar-provider";
-import { TaxProvider } from "./tax-provider";
+import { AvataxWebhookService } from "../avatax/avatax-webhook.service";
+import { TaxJarWebhookService } from "../taxjar/taxjar-webhook.service";
+import { ProviderWebhookService } from "./tax-provider-webhook";
 import { TaxProviderError } from "./tax-provider-error";
 import pino from "pino";
+import { getAppConfig } from "../app-configuration/get-app-config";
 
-export class ActiveTaxProvider {
-  private client: TaxProvider;
+type ActiveTaxProviderResult =
+  | { ok: true; data: ActiveTaxProvider }
+  | { ok: false; data: null; error: string };
```

Comment:
why you model `data: null` here? Its perfectly fine and more natural to *not* declare data in Result with ok: false